### DISCUSSION TITLE
Using `get` to retrieve store values doesn't trigger a component re-render

### DISCUSSION THREAD

**Discussion by vcardins** [udecode/zustand-x#87] [Q&A] [answered]
**Description**

I might be using it wrongly, so excuse me in this case ...
Using `get` to retrieve store values doesn't trigger a component re-render

**Steps**

1. Create a component which listen to values
2. Update store values in a different component/provider, etc
3. Check whether the components which listen/consumes to the store value
4. See error --> Notice it doesn't get updated

If I use `get` ... no re-renders

```
export const ParametersControl = memo(() => {
	const { set, get } = dataStore;

	const collection = get.collection?.();
	const collections = get.collections();
	const colorMap = get.colorMap();
	const time = get.time?.();
```
![image](https://github.com/user-attachments/assets/d3f80ad0-150b-479b-93be-59a17b7701d2)

If I use `use` ... it re-renders

```
export const ParametersControl = memo(() => {
	const { set, use } = dataStore;

	const collection = use.collection?.();
	const collections = use.collections();
	const colorMap = use.colorMap();
	const time = use.time?.();
```
![image](https://github.com/user-attachments/assets/2a8c00b7-e36c-4058-95bb-5a8cde2c0365)

**Expectation**

Should get and use work the same way? What am I missing?

**Comment by zbeyens** [comment]
`use` is a hook, `get` is not. The latter can be used in callbacks or outside React.

**Reply by vcardins** [reply]
Thanks for the prompt response. So, basically what you're saying is: in react components I should use `use` instead of `get`. Is it?

**Reply by zbeyens** [reply, marked as answer]
Yes, if you want to re-render on change, which is not always necessary (e.g. in a React callback).
